### PR TITLE
use ubuntu 22.04 and rockylinux 9 as base images

### PR DESCRIPTION
**Pull request by glyg**
There is still an issue with `lsb_release` not being provided by rockylinux

Searching for the command [on the whole org](https://github.com/search?q=org%3Aome%20lsb_release&type=code) does not return anything outside of the ubuntu world.

I can try advancing the downstream roles to see whether this is an issue for deployments

### CODE REVIEW

**Comment by sbesson** [comment]
GitHub Actions builds are deactivated due to the 60 days inactivity and will need to be reenabled /cc @pwalczysko @jburel 

Re [6707e75](https://github.com/ome/ansible-role-basedeps/pull/12/commits/6707e753c515015696d5356f8b380085f5c654aa), the installation of LSB core packages comes from https://github.com/ome/ansible-role-basedeps/commit/d0f80310d4cd712b3c071f82e1894c81e8895ef1. Given the state of the LSB project, my vote would be for completely getting rid of it as part of this work - https://lists.linuxfoundation.org/pipermail/lsb-discuss/2023-February/008278.html

**Comment by JulianHn** [comment]
I would not rely on `lsb_release` in any new code if possible. The LSB project is basically dead since a few years (as also stated by one of its maintainers https://lists.linuxfoundation.org/pipermail/lsb-discuss/2023-February/008278.html) and is suspect more and more distros will drop the `lsb_release` command. 
Probably a good idea to start exchanging the `lsb_release` commands for parsing `/etc/os-release`. That is AFAIC provided with all major server distros (Ubuntu, SLES, RedHat, Debian).

**Comment by joshmoore** [comment]
> GitHub Actions builds are deactivated due to the 60 days inactivity and will need to be reenabled /cc @pwalczysko @jburel

Done

**Comment by glyg** [comment]
I'm pruning lsb_release then :)

File: molecule/default/molecule.yml [hunk 1]
```diff
@@ -8,10 +8,10 @@ lint: |
     ansible-lint
     flake8
 platforms:
-  - name: centos-7
-    image: centos/systemd
-  - name: ubuntu-2004
-    image: ubuntu:20.04
+  - name: rockylinux
```
Comment (sbesson) [inline molecule/default/molecule.yml]: Here (and in https://github.com/ome/ansible-role-java/pull/14), my inclination would be to include the major version of the distribution in the name i.e. `rockylinux9` or `rockylinux-9` to be unambiguous.

Comment (jburel) [inline molecule/default/molecule.yml]: we may want to be more explicit i.e. rockylinux9 to avoid confusion

Comment (jburel) [reply molecule/default/molecule.yml]: agree

**Review by sbesson — CHANGES_REQUESTED** [review]
Looking great @glyg. A few comments:

- with the recent Molecule upgrade (https://github.com/ome/ansible-role-basedeps/pull/11), I think it might be possible to get rid of `molecule/default/Dockerfile.j2`. Can you give it a try?
- https://github.com/ome/ansible-role-basedeps/blob/5d2c034f605255eae465e3ff31cd5b6dc1ac5014/meta/main.yml#L10-L17 also needs to be updated to reflect the platform support
- is there a particular requirement for setting `allowerasing` to `true`  in the RedHat installation?

**Comment by glyg** [comment]
> is there a particular requirement for setting allowerasing to true in the RedHat installation?

Yes it is necessary for curl, because `crul-minimal` is already installed on rockylinux-9. The alternative would be to not install curl on rockylinux (so with a tad more code)


> with the recent Molecule upgrade (https://github.com/ome/ansible-role-basedeps/pull/11), I think it might be possible to get rid of molecule/default/Dockerfile.j2. Can you give it a try?

Is this the correct link?

**Comment by pwalczysko** [comment]
> > with the recent Molecule upgrade (#11), I think it might be possible to get rid of molecule/default/Dockerfile.j2. Can you give it a try?
> 
> Is this the correct link?

@glyg No, sorry, it is not, because in that particular case I did not manage to get rid of the Dockerfile. Please have a look maybe at https://github.com/ome/ansible-role-java/pull/13 and namely on https://github.com/ome/ansible-role-java/pull/13/commits/d9d83c54aed6e68a78d3d2f2dec5afd1037d9ef8 - as you probably gathered already, this was a bit of a trial and error - the Dockerfiles sometimes contain installations which, with my modest knowledge of the topic, I was not able to replace otherwise, and was forced to leave the Dockerfiles in such cases.

**Comment by glyg** [comment]
@pwalczysko Trying to remove the dockerfile (or the dnf part of it) failed with a `sudo not found` error at the `install epel repo` stage. I must say I don't understand why ansible would need sudo here to become root, and not on the ansible-role-java playbook.

**Comment by glyg** [comment]
My bad, actually removing the dockerfile entirely and updating apt cache for ubuntu seems ok

**Comment by pwalczysko** [comment]
> failed with a `sudo not found` error at the `install epel repo` stage.

Indeed, this type of error used to stop me, and I opted for the "dirty" solution of leaving the Dockerfile in.



> My bad, actually removing the dockerfile entirely and updating apt cache for ubuntu seems ok

Brilliant, and your test is passing too, thank you for sorting that one out.

**Comment by sbesson** [comment]
> Yes it is necessary for curl, because crul-minimal is already installed on rockylinux-9. The alternative would be to not install curl on rockylinux (so with a tad more code)

😢 Without too much thought, the second approach would have my personal preference. I am unaware of a requirement to use functionalities of `curl` beyond those available in the minimal package. In that sense, allowing this role replacing the base packages coming with EL9 distributions feels intrusive and unnecessary.
@joshmoore @jburel @pwalczysko @khaledk2 additional opinions welcome

**Comment by glyg** [comment]
> In that sense, allowing this role replacing the base packages coming with EL9 distributions feels intrusive and unnecessary.

Yes, and also contrary to a "minimal attack surface" philosophy.

**Comment by glyg** [comment]
What do you think of this solution?
In the longer run, I feel it might be easier to maintain

**Review by sbesson — APPROVED** [review]
Thanks. As discussed earlier in the thread, proposed set of changes look reasonable to me both in terms of features and clean-up. Passing off to the OME  team for a review on their side. 

An upcoming decision is whether we start releasing these updated roles as we fix them, likely with a major version increment to communicate the breaking platform requirement. (NB: we could also bump the minimal Ansible version).

**Review by jburel — APPROVED** [review]

**Pull request merged**